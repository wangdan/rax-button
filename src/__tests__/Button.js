import {createElement, Component} from 'rax';
import renderer from 'rax-test-renderer';
import Button from '../';

class ButtonTest extends Component {
  state = {
    buttonText: 'normal'
  };
  render() {
    return <Button onPress={() => {
      this.setState({
        buttonText: 'click'
      });
    }}>{this.state.buttonText}</Button>;
  }
}
describe('Button', () => {
  it('should render a button', () => {
    const component = renderer.create(
      <ButtonTest />
    );
    let tree = component.toJSON();

    expect(tree.tagName).toEqual('DIV');
    expect(tree.children[0].children[0]).toEqual('normal');
    expect(tree).toMatchSnapshot();

    tree.eventListeners.click();

    tree = component.toJSON();
    expect(tree.children[0].children[0]).toEqual('click');
  });
});
