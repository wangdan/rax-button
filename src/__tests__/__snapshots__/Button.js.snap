// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
Object {
  "attributes": Object {
    "role": "button",
  },
  "children": Array [
    Object {
      "children": Array [
        "normal",
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "block",
        "flexDirection": "column",
        "flexShrink": 0,
        "fontSize": 32,
        "position": "relative",
        "whiteSpace": "pre-wrap",
      },
      "tagName": "SPAN",
    },
  ],
  "eventListeners": Object {
    "click": [Function],
  },
  "style": Object {
    "alignContent": "flex-start",
    "border": "0 solid black",
    "boxSizing": "border-box",
    "cursor": "pointer",
    "display": "flex",
    "flexDirection": "column",
    "flexShrink": 0,
    "position": "relative",
  },
  "tagName": "DIV",
}
`;
